src/components/datagrid/data_grid.tsx

```diff
@@ -598,37 +598,29 @@ export const EuiDataGrid: FunctionComponent<EuiDataGridProps> = props => {
     </EuiI18n>
   );
 
-  const [cellsUpdateFocus, setCellsUpdateFocus] = useState<
-    Array<Function[] | null[]>
-  >([]);
+  const [cellsUpdateFocus] = useState<Map<string, Function>>(new Map());
```

Comment:
Instead of the nested arrays to track `onFocusUpdate` listeners, I refactored to be a map. I think this results in more readable code - I quickly get lost when working with nested arrays and all those `[` `]` brackets.